### ISSUE TITLE
Drop suffix viewController for subclasses of UIViewController

### ISSUE THREAD

**Issue by 3lvis** [hyperoslo/iOS-playbook#130]
I was thinking that we could just use `controller`. For example you have an `HYPRecipesViewController`, you would only name it `HPYRecipesController`or in Swift just `RecipesController`. 

Is not that for subclasses of UITableViewController you do `RecipesTableViewController`, and even if you did that it's not a requirement since knowing the parent class isn't super difficult, meanwhile writing the super long name everywhere can be quite painful.

-----

 tl;dr `HYPRecipesTableViewController` => `HYPRecipesController`

What do you guys think @hyperoslo/ios?

**Comment by attila-at-hyper** [comment]
:+1:

**Comment by kostiakoval** [comment]
How about data controller, not a view controller, how would you name it?

**Comment by kostiakoval** [comment]
ImageController, FetchingController, etc

**Comment by 3lvis** [comment]
What's a data controller?

**Comment by 3lvis** [comment]
What would a ImageController do? Or a FetchingController?

**Comment by kostiakoval** [comment]
HYPRecipesController for example could download and validate In-APP Purchase receipts

**Comment by kostiakoval** [comment]
HYPRecipesController: NSObject - validate verecepts. 
Would use use controller suffix only for ViewControllers ?

**Comment by 3lvis** [comment]
I wouldn't name something like that a `HYPRecipesController`.

**Comment by 3lvis** [comment]
>Would use use controller suffix only for ViewControllers? 

I had that in mind. Usually anything else called `Controller` is usually code smell.

**Comment by kostiakoval** [comment]
Ok

**Comment by zenangst** [comment]
`Controller`'s and `Manager`'s beware, you smell

**Comment by RamonGilabert** [comment]
I used Manager, it's bad? I agree with the Issue @3lvis! :)

**Comment by vadymmarkov** [comment]
Agree with @3lvis to drop `ViewController` suffix, but we could be in conflict with Apple's naming then. For example `NSFetchedResultsController` is not view controller, but has `Controller` suffix.
As for managers I think `Manager` is too generic, I would name it something like DataFetcher instead of `DataFetchManager`, but it's another discussion I guess.

**Comment by 3lvis** [comment]
The naming of `NSFetchedResultsController` is just unfortunate, wouldn't take it as an example of good naming. It could be called `NSResultsFetcher` or something else.

**Comment by vadymmarkov** [comment]
Yes, `NSResultsFetcher` is much better, but I mean that Apple did it for some reason and we can't change it :smile: .If you think that it's not very important to follow some Apple's name conventions, then we are Ok.

**Comment by 3lvis** [comment]
They have been quite inconsistent too.

UITabBarController, UISearchController, UINavigationController, UIAlertController, MPMediaPickerController, EKCalendarChooser for example, all of them subclass UIViewController.

**Comment by 3lvis** [comment]
@hyperoslo/ios guys => https://github.com/hyperoslo/iOS-playbook/pull/131
